src/pal/src/thread/thread.cpp

```diff
@@ -1578,7 +1578,117 @@ GetThreadTimes(
     return (retval);
 }
 
+HRESULT
+PALAPI
+SetThreadName(    
+    IN HANDLE hThread,
+    IN PCWSTR lpThreadDescription)
+{
+    CPalThread *pThread;
+    PAL_ERROR palError;
+
+    PERF_ENTRY(SetThreadName);
+    ENTRY("SetThreadName(hThread=%p,lpThreadDescription=%p)\n", hThread, lpThreadDescription);
+
+    pThread = InternalGetCurrentThread();
+
+    palError = InternalSetThreadName(
+        pThread,
+        hThread,
+        lpThreadDescription
+        );
+
+    if (NO_ERROR != palError)
+    {
+        pThread->SetLastError(palError);
+    }
+
+    LOGEXIT("SetThreadNameExit");
+    PERF_EXIT(SetThreadName);
+
+    return NO_ERROR != palError ? 0x80004005 : 0; // E_FAIL
```

Comment:
I wasn't sure how to correctly get a hold of the HRESULT macros, or if I should even return a bog standard one like E_FAIL here.

Reply:
The problem is that we don't use HRESULT anywhere in PAL, so the macros to translate between win32 error (which is the same as PAL error) and HRESULT are not available. We have a couple of options:
1. Move the https://github.com/dotnet/coreclr/blob/921b39ccb5d075cb61ee9b167d0d17fc4acfda59/src/pal/inc/rt/palrt.h#L321-L373 to the pal.h, which would allow you to use the `HRESULT_FROM_WIN32(palError)` here.
2. Change the function to return PAL_ERROR instead of HRESULT and do the translation to HRESULT in the wrapper function, I've mentioned in my previous comment. In that case, we should rename the function to have a `PAL_` prefix to indicate it is not a 1:1 implementation of the Windows API.

I would prefer option 1